jupyter_kernel_mgmt/tests/test_client_loop.py

```diff
@@ -3,95 +3,101 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 
-
 import os
+import pytest
 try:
     from queue import Queue, Empty
 except ImportError:
     from Queue import Queue, Empty
 
 pjoin = os.path.join
-from unittest import TestCase
 
+from async_generator import yield_, async_generator
 from ipykernel.kernelspec import make_ipkernel_cmd
 from jupyter_protocol.messages import Message
-from jupyter_kernel_mgmt.subproc import SubprocessKernelLauncher
+from jupyter_kernel_mgmt.subproc.async_manager import (
+    AsyncSubprocessKernelLauncher, start_new_kernel
+)
 from jupyter_kernel_mgmt.client import ClientInThread
-from .utils import test_env
 
 from ipython_genutils.py3compat import string_types
 
 TIMEOUT = 30
 
+pytestmark = pytest.mark.asyncio
+
+@pytest.fixture
+@async_generator
+async def setup_test(setup_env):
+
+    # Start a client in a new thread, put received messages in queues.
+    launcher = AsyncSubprocessKernelLauncher(make_ipkernel_cmd(), cwd='.')
+    connection_info, km = await launcher.launch()
+    pytest.kc = ClientInThread(connection_info, manager=km)
```

Comment:
Why are we attaching things to the `pytest` module? The usual way to write tests with pytest fixtures is something like this:

```python
@pytest.fixture
def client_in_thread(setup_env):
    kc = ClientInThread(...)
    ...
    yield kc
    ...

def test_something(client_in_thread):
    client_in_thread.complete(...)
```

This avoids modifying global state. If you need to pass multiple objects to the test, you can use a tuple or a dict - not quite as elegant, but still better than modifying the pytest module.

Reply:
Just coming up to speed on pytest - thanks for the pointers, didn't realize modifying pytest was bad form.  I think I need to yield a dict since `received` is part of this.